### PR TITLE
CNF-11569: update testConfig to handle GM configuration

### PR DESCRIPTION
**Pull request by nishant-parekh**

### CODE REVIEW

File: test/pkg/ptphelper/ptphelper.go [hunk 1]
```diff
@@ -666,3 +667,111 @@ func IsExternalGM() (out bool) {
...
+	}
+	return resMap
+}
+
+func checkGNSSAvailabilityForIface(nodeName string, IfaceName string) (string, bool) {
+	cmd := []string{"/bin/sh", "-c", fmt.Sprintf("ls /sys/class/net/%s/device/gnss", IfaceName)}
+	logrus.Infof("cmd = %s ", cmd)
+	so, se, err := execPodCommand(nodeName, cmd)
+	if err != nil {
+		logrus.Errorf("could not gnss device, err: %s \n stderr: %s interfaceName %s", err, se.String(), cmd)
```
Comment (jzding) [inline test/pkg/ptphelper/ptphelper.go:723]: `cmd` is passed to `interfaceName %s`, shouldn't it be `IfaceName`? Also print out `nodeName` to help debug.

File: test/pkg/ptphelper/ptphelper.go [hunk 2]
```diff
@@ -666,3 +667,111 @@ func IsExternalGM() (out bool) {
...
+	return false
+}
+
+func execPodCommand(nodeName string, cmd []string) (stdoutBuf, stderrBuf bytes.Buffer, err error) {
+
+	WaitForPtpDaemonToBeReady()
+	pod, err := GetPtpPodOnNode(nodeName)
+	so, se := bytes.Buffer{}, bytes.Buffer{}
+	if err != nil {
+		logrus.Errorf("Could not get ptp pod from node due to err: %s, %s", err, nodeName)
```
Comment (jzding) [inline test/pkg/ptphelper/ptphelper.go:769]: ("Could not get ptp pod from node %s, %s", nodeName, err)

File: test/pkg/testconfig/testconfig.go [hunk 3]
```diff
@@ -974,11 +1143,66 @@ func PtpConfigDualNicBC(isExtGM bool) error {
...
+		// Check GM interface available
+		grandmaster = (*data.testClockRolesAlgoMapping[BestSolution])[Grandmaster]
+		gmIf := GlobalConfig.L2Config.GetPtpIfList()[(*data.solutions[BestSolution])[FirstSolution][grandmaster]]
+
+		// Check the Iface has a WPC NIC associated to it
+		deviceID, IfList, hasWPC := GetWPCNICIface(gmIf.NodeName, gmIf.IfName)
+
+		if !hasWPC && len(IfList) == 0 {
+			logrus.Error("WPC NIC not found in list of interfaces on the cluster")
+			return fmt.Errorf("WPC NIC not found in list of interfaces on the cluster %d", len(IfList))
```
Comment (jzding) [inline test/pkg/testconfig/testconfig.go:1164]: ("WPC NIC not found in list of %d interfaces on the cluster", len(IfList))

File: test/pkg/testconfig/testconfig.go [hunk 4]
```diff
@@ -1092,3 +1324,19 @@ func discoverMode(ptpConfigClockUnderTest []*ptpv1.PtpConfig) {
 	}
 	GlobalConfig.DiscoveredClockUnderTestPod = pod
 }
+
+// Helper function to identify WPC NIC
+func GetWPCNICIface(nodeName string, solIfName string) (string, []string, bool) {
```
Comment (jzding) [inline test/pkg/testconfig/testconfig.go:1329]: I don't see any need of this function. You can just call ptphelper.GetListOfWPCEnabledInterfaces(nodeName) directly.

File: test/conformance/serial/ptp.go [hunk 5]
```diff
@@ -824,6 +830,23 @@ var _ = Describe("["+strings.ToLower(DesiredMode.String())+"-serial]", Serial, f
...
 		})
+
+		Context("WPC Grandmaster tests", func() {
+			BeforeEach(func() {
+				By("Refreshing configuration", func() {
+					ptphelper.WaitForPtpDaemonToBeReady()
+					fullConfig = testconfig.GetFullDiscoveredConfig(pkg.PtpLinuxDaemonNamespace, true)
+				})
+				if fullConfig.Status == testconfig.DiscoveryFailureStatus {
+					Skip("Failed to find a valid ptp slave configuration")
```
Comment (jzding) [inline test/conformance/serial/ptp.go:841]: slave -> master

**Review by aneeshkp — APPROVED** [review]
/lgtm

File: test/conformance/serial/ptp.go (hunk 5 above)
Comment (nishant-parekh) [reply test/conformance/serial/ptp.go]: fixed

**Review by jzding — APPROVED** [review]
/lgtm

**Pull request merged**